griptape/clients/griptape_cloud_api_client.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+import os
+from abc import ABC
+from typing import Optional
+
+import requests
+from attrs import Factory, define, field
+
+from griptape.utils import dict_merge
+
+
+@define
+class GriptapeCloudApiClient(ABC):
+    base_url: str = field(
+        default=Factory(lambda: os.getenv("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")),
+        kw_only=True,
+    )
+    api_key: Optional[str] = field(default=Factory(lambda: os.getenv("GT_CLOUD_API_KEY")), kw_only=True)
+    headers: dict = field(
+        default=None,
+        kw_only=True,
+    )
+    structure_run_id: Optional[str] = field(
+        default=Factory(lambda: os.getenv("GT_CLOUD_STRUCTURE_RUN_ID")), kw_only=True
+    )
+
+    def __attrs_post_init__(self) -> None:
+        if self.api_key is None:
+            raise ValueError("GriptapeCloudApiClient requires an API key")
+        self.headers = dict_merge(
+            {"Authorization": f"Bearer {self.api_key}"},
+            self.headers if self.headers is not None else {},
+        )
```

Comment:
Do we want to merge this in? What if they provide a different key in their `headers`? Definitely, but this should probably be renamed something like `extra_headers` if we're going to add them in.

Reply:
i feel like its not intended to break the client if you want to add headers. i dont event really know why we provide the option to begin with because the API doesnt accept extra headers. if you want to give your own `Authorization` itll override, else itll just add it in

Reply:
made headers `init=False`. theres no reason to give headers at this time.